docs: improve Next.js integration guide with comprehensive examples

adarsh-priydarshi-5646 opened the pull request:
- Add Quick Start section with connection caching example
- Add Best Practices section covering connection caching, environment variables, and model registration
- Add Pages Router examples for API routes and getServerSideProps
- Add App Router Server Components example
- Add Additional Resources section with helpful links
- Reorganize content for better readability

Fixes #6587

**Summary**

This PR significantly improves the Next.js integration documentation by adding comprehensive examples and best practices. The existing documentation was minimal and didn't cover common use cases that developers encounter when integrating Mongoose with Next.js, particularly around connection management in serverless environments and proper usage patterns for both Pages Router and App Router. This adds a Quick Start with a connection caching pattern, a Best Practices section, practical examples for both the Pages Router and the App Router, and Additional Resources with links to official examples.

**Examples**

Connection caching pattern for serverless environments:

```let cached = global.mongoose;
if (!cached) {
  cached = global. mongoose = { conn: null, promise: null };
}
async function dbConnect() {
  if (cached.conn) return cached.conn;
  if (!cached.promise) {
    cached.promise = mongoose.connect(MONGODB_URI, opts);
  }
  cached.conn = await cached.promise;
  return cached.conn;
}
```

Model registration pattern to prevent hot reload errors:

export default mongoose.models. User || mongoose.model('User', UserSchema);

Using with getServerSideProps:

```export async function getServerSideProps() {
  await dbConnect();
  const users = await User. find({}). lean();
  return { props: { users: JSON.parse(JSON.stringify(users)) } };
}
```

vkarpov15 commented on docs/nextjs.md:
```diff
@@ -5,6 +5,105 @@ Mongoose works out of the box with Next.js.
+  cached = global.mongoose = { conn: null, promise: null };
+}
+
+async function dbConnect() {
```
Overall this entire function could be drastically simplified by having something like `if (!cached) { cached = mongoose; await mongoose.connect(); }`

adarsh-priydarshi-5646 commented:
@vkarpov15, Thanks for the review! I've simplified the dbConnect() function and fixed all the issues you mentioned. Changed @/ to relative imports, added runtime = 'nodejs' to the App Router example, and made _id.toString() consistent across all examples. Also fixed the markdown lint errors.
